packages/react-reconciler/src/ReactFiberTreeReflection.js

```diff
@@ -342,20 +342,52 @@ export function isFiberSuspenseAndTimedOut(fiber: Fiber): boolean {
   );
 }
 
-// This is only safe to call in the commit phase when the return tree is consistent.
-// It should not be used anywhere else. See PR #18609 for details.
-export function isFiberInsideHiddenOrRemovedTree(fiber: Fiber): boolean {
-  let node = fiber;
-  let lastChild = null;
+function doesFiberContain(parentFiber: Fiber, childFiber: Fiber): boolean {
+  let node = childFiber;
+  const parentFiberAlternate = parentFiber.alternate;
   while (node !== null) {
-    if (
-      node.effectTag & Deletion ||
-      (isFiberSuspenseAndTimedOut(node) && node.child === lastChild)
-    ) {
+    if (node === parentFiber || node === parentFiberAlternate) {
       return true;
     }
-    lastChild = node;
     node = node.return;
   }
   return false;
 }
+
+function isFiberHiddenAndContains(
```

Comment:
It’s a bit unclear to me whether which fiber we’re testing for whether it’s hidden from the name.

Reply:
Updated functions to better reflect intent.